Sources/NIOCore/EventLoopFuture.swift

```diff
@@ -1825,13 +1949,19 @@ public struct _NIOEventLoopFutureIdentifier: Hashable, Sendable {
     }
 }
 
-// EventLoopPromise is a reference type, but by its very nature is Sendable (if its Value is).
-extension EventLoopPromise: Sendable where Value: Sendable { }
+// The promise and future are both unchecked Sendable following the below isolation rules this is safe
+//
+// 1. Receiving the value of the future is always done on the EventLoop of the future, hence
+// the value is never transferred out of the event loops isolation domain. It only gets transferred
+// by certain methods such as `hop()` and those methods are annotated with requiring the Value to be
+// Sendable
+// 2. The promise is `Sendable` but fulfilling the promise with a value requires the user to
+// transfer the value to the promise. This ensures that the value is now isolated to the event loops
+// isolation domain. Note: Sendable values can always be transferred
+
+extension EventLoopPromise: @unchecked Sendable { }
```

Comment:
Non need for unchecked. `EventLoopFuture` is already `Sendable` and that is the only store property.
```suggestion
extension EventLoopPromise: Sendable { }
```

Reply:
I think this still applies.